packages/notebook-extension/src/index.ts

```diff
@@ -491,12 +498,66 @@ const trusted: JupyterFrontEndPlugin<void> = {
   },
 };
 
+/**
+ * Add a command to open right sidebar for Editing Notebook Metadata when clicking on "Edit Notebook Metadata" under Edit menu
+ */
+const editNotebookMetadata: JupyterFrontEndPlugin<void> = {
+  id: '@jupyter-notebook/notebook-extension:edit-notebook-metadata',
+  description:
+    'Add a command to open right sidebar for Editing Notebook Metadata when clicking on "Edit Notebook Metadata" under Edit menu',
+  autoStart: true,
+  optional: [ITranslator, INotebookTools],
+  activate: (
+    app: JupyterFrontEnd,
+    translator: ITranslator | null,
+    notebookTools: INotebookTools | null
+  ) => {
+    const { commands } = app;
+    translator = translator ?? nullTranslator;
+    const trans = translator.load('notebook');
+
+    commands.addCommand(CommandIDs.openEditNotebookMetadata, {
+      label: trans.__('Edit Notebook Metadata'),
+      execute: async () => {
+        const command = 'application:toggle-panel';
+        const args = {
+          side: 'right',
+          title: 'Show Notebook Tools',
+          id: 'notebook-tools',
+        };
+
+        // Check if Show Notebook Tools (Right Sidebar) is open (expanded)
+        if (!commands.isToggled(command, args)) {
+          await commands.execute(command, args).then((_) => {
+            // For expanding the 'Advanced Tools' section (default: collapsed)
+            if (notebookTools) {
+              const tools = (notebookTools?.layout as any).widgets;
+              tools.forEach((tool: any) => {
+                if (
+                  tool.widget.title.label === trans.__('Advanced Tools') &&
```

Comment:
This looks a little brittle. But I guess there is no other way to programmatically expand the "Advanced Tools" section?

Reply:
that's true @jtpio, back while writing, I tried to find simpler and yk, more streamlined way of doing it but couldn't find any and functionally expanding 'Advanced Tools' section was necessary so couldn't avoid it.

Reply:
Right, maybe we can stick to this for now then.

Reply:
yes